ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/BlockProcessor.java

```diff
@@ -60,7 +63,8 @@ BeaconState processAndValidateBlock(
       SignedBeaconBlock signedBlock,
       BeaconState blockSlotState,
       IndexedAttestationCache indexedAttestationCache,
-      Optional<? extends OptimisticExecutionPayloadExecutor> payloadExecutor)
+      Optional<? extends OptimisticExecutionPayloadExecutor> payloadExecutor,
+      Function<SlotAndBlockRoot, Optional<List<InclusionList>>> inclusionListSupplier)
```

Comment:
Is there a reason for using a supplier here? We could directly get the ILs as `Optional<List<InclusionList>>` from the store since we have the slot and block root when running a fork choice on a block (The `ForkChoice::onBlock` has the slot and the block root

Reply:
So my choice to use the supplier here was more in terms of making sure that we only try to fetch the IL when we're assembly the payload, but I think at this stage the block has been already formed so yeah we can probably just retrieve the IL at `ForkChoice::onBlock`. Thanks for the review